### PR TITLE
feat: add owlbot api changes to auto-approve

### LINKED ISSUE #2298: feature-request(auto-approve): autoapprove upstream-API-changes that owl-bot proposes
Per our conversation with @parthea, we have a provisional sense of what these requirements would entail:

1. No breaking changes
2. If there are other owl-bot PRs in the repo, don't autoapprove (so as to not merge out-of-order)
3. Must have a piper-origin ID
4. Must not have any other commits from other sources.

We also have an idea to autoapprove template changes: #2297

### PR DESCRIPTION
**Pull request by sofisl**
Fixes #2298

### CODE REVIEW

File: packages/auto-approve/src/process-checks/update-discovery-artifacts.ts [hunk 1]
```diff
@@ -63,7 +61,6 @@ export class UpdateDiscoveryArtifacts extends Process implements LanguageRule {
       (this.classRule = {
         author: 'yoshi-code-bot',
         titleRegex: /^chore: Update discovery artifacts/,
-        maxFiles: 2,
```
Comment (sofisl) [inline packages/auto-approve/src/process-checks/update-discovery-artifacts.ts:66]: @parthea, saw a bug with the update-discovery-artifacts process, currently it caps the amount of files changed to 2, but your PRs are usually much bigger. I removed it in this PR.

File: packages/auto-approve/README.md [hunk 2]
```diff
@@ -98,6 +100,14 @@ Below is what each process checks for:
   - Checks that the title of the PR does NOT include feat, fix, or !
   - Checks that the body of the PR does not contain 'PiperOrigin-RevId'
   - Checks that the .repo-metadata.json of the repo contains "library_type": "GAPIC_AUTO"
+* OwlBotAPIChanges:
+  - Checks that the author is 'gcf-owl-bot[bot]'
+  - Checks that the title of the PR does NOT include breaking, BREAKING, or !
+  - Checks that the body of the PR DOES contain 'PiperOrigin-RevId'
+  - Checks that the .repo-metadata.json of the repo contains "library_type": "GAPIC_AUTO"
+  - Checks that there are no other PRs in that repository that have been opened by gcf-owl-bot[bot]
```
Comment (bcoe) [inline packages/auto-approve/README.md:108]: Good call with this constraint :smile:

File: packages/auto-approve/src/get-pr-info.ts [hunk 3]
```diff
@@ -165,3 +165,18 @@ export async function getFileContent(
...
   );
 }
+
+export async function listCommitsOnAPR(
+  owner: string,
+  repo: string,
+  prNumber: number,
+  octokit: Octokit
+) {
+  const commits = await octokit.paginate(await octokit.pulls.listCommits, {
```
Comment (bcoe) [inline packages/auto-approve/src/get-pr-info.ts:175]: will this auto-paginate all commits?

**Review by bcoe — APPROVED** [review]
Great job with your refactor, these new rules read well and are easy to follow :+1:

**Review by parthea — APPROVED** [review]

File: packages/auto-approve/src/get-pr-info.ts (hunk 3 above)
Comment (sofisl) [reply packages/auto-approve/src/get-pr-info.ts:175]: yes, it should!

**Pull request merged**